src/depackers/vorbis.c

```diff
@@ -3549,7 +3540,7 @@ static int vorbis_pump_first_frame(stb_vorbis *f)
    int len, right, left, res;
    res = vorbis_decode_packet(f, &len, &left, &right);
    if (res)
-      res = vorbis_finish_frame(f, len, left, right);
+      vorbis_finish_frame(f, len, left, right);
```

Comment:
An old libxmp-specific hack (https://github.com/libxmp/libxmp/commit/fb4b9dda1deb864e64ed1a862767870b5eedbd90) was removed in #408 that could cause `vorbis_finish_frame` to return -1. This was obsoleted by an upstream check that was added that returns 0 instead.

Based on upstream usage, `vorbis_pump_first_frame` should not be returning the return value of `vorbis_finish_frame` (which is the number of samples written), but instead a `TRUE` or `FALSE` value. To add to the confusion, at some point the libxmp copy of stb_vorbis was modified to use 0/-1 return values, so a later check in deadcode incorrectly returns NULL when the return value of this is NOT zero. I've just reverted this entire mess. This includes one of the two checks in `stb_vorbis_get_frame_float` that have caused a lot of trouble elsewhere.